golang/1.9/go-test-coverage-lint

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+mkdir -p /coverage
+mkdir -p /log
+
+if [ "$#" -eq 0 ]; then
+	PKG="./..."
+else
+	PKG=${@:1}
+fi
+
+function test-coverage {
+    echo "***********"
+    echo "* go test *"
+    echo "***********"
+
+    # Capture the test result
+    set -o pipefail
+    go test -v $PKG | go2xunit > /log/test.xml
+    code=$?
+    set +o pipefail
+
+    echo "mode: count" > /coverage/coverage.out
+
+    # Generate coverage for our source files
+    for x in $(go list $PKG | grep -v /vendor/); do
+    	file=/coverage/$(echo $x | tr / -)
+    	go test -covermode=count -coverprofile=$file "$x"
```

Comment:
Does `go test -covermode` support multiple files in Go 1.9?

Reply:
Yup

Reply:
Oh in that case, we can simplify this for loop? The reason we needed to use this for loop was because covermode don't let you specify the packages. 

I think it lets you do all package, and if in Go 1.9 `./...` ignores vendor, it should work.

Reply:
Seems like `coverprofile` does not support files over several packages. `cannot use test profile flag with multiple packages`